src/solvers/smt2/smt2_solver.cpp

```diff
@@ -166,7 +172,9 @@ void smt2_solvert::command(const std::string &c)
     {
       // this is a command that Z3 appears to implement
       exprt e=expression();
-      if(e.is_not_nil())
+      if(!ok)
+        ok = true;
```

Comment:
What's going on in all these "if it's not ok then it's ok" statements?

Reply:
will drop.